client/src/components/EditSurveyForm.js

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+
+import {
+  Form,
+  FormLayout,
+  TextField,
+  Button,
+  Card,
+  Stack
+} from "@shopify/polaris";
+
+export function EditSurveyForm({
+  surveyId,
+  surveyTitle,
+  handleSurveyTitleChange,
+  questions = [],
+  handleQuestionChange
+}) {
+  const handleSubmit = async () => {
+    const updateTitleRequest = fetch(`/api/surveys/${surveyId}`, {
+      method: "PUT",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({ name: surveyTitle })
+    });
+
+    function questionRequestHelper(question, body) {
+      return fetch(`/api/questions/${question.id}`, {
+        method: "PUT",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify(body)
+      });
+    }
+
+    const questionRequests = questions
+      .map(question => {
+        if (question.type === "text") {
+          return questionRequestHelper(question, {
+            type: "text",
+            question: question.question
+          });
+        }
+        if (question.type === "range") {
+          return questionRequestHelper(question, {
+            type: "range",
+            question: question.question,
+            min: question.min,
+            max: question.max
+          });
+        }
+        if (question.type === "option") {
+          return questionRequestHelper(question, {
+            type: "option",
+            question: question.question,
+            options: question.options
+          });
+        }
+
+        return null;
+      })
+      .filter(question => !!question);
+
+    await Promise.all([updateTitleRequest, questionRequests]);
+    window.location = "/app/admin/surveys";
+  };
+
+  const questionsMarkup = questions.map((question, index) => {
```

Comment:
is it worth making a component to share the code with question markup from `Survey.js`?

Reply:
Ideally, yes.